src/components/MagicCodeInput.js

```diff
@@ -181,9 +233,12 @@ function MagicCodeInput(props) {
             return;
         }
 
+        const addedValue =
```

Comment:
Can you add a comment to explain this? It's very confusing to look at, haha. Also some parentheses around the boolean section might help.